experimental/sdk/webpubsub-socketio-extension/src/SIO/components/web-pubsub-adapter.ts

```diff
@@ -82,8 +83,26 @@ export class WebPubSubAdapterInternal extends NativeInMemoryAdapter {
    * @param opts - the filters to apply
    * @param rooms - the rooms to join
    */
-  public addSockets(opts: BroadcastOptions, rooms: Room[]): void {
-    throw NotImplementedError;
+  public async addSockets(opts: BroadcastOptions, rooms: Room[]): Promise<void> {
+    debug(`addSockets, start, rooms = ${toString(rooms)}, opts = ${toOptionsString(opts)}`);
+    const localSockets = await super.fetchSockets(opts);
+    const releases: MutexInterface.Releaser[] = [];
+    localSockets.forEach(async (socket) => {
```

Comment:
Do we need lock for this method as seems it's not directly used by socketio codes. And this lock may cause dead lock as you're acquiring some locks when you hold some locks

Reply:
removed lock for `addSockets` and `delSockets`